HashRegistrarSimplified.sol

```diff
@@ -92,23 +92,49 @@ contract Deed {
     function () payable {}
 }
 
+
+/**
+ * @title Previous Registrar
+ * @dev The old registrar interface, for an upgrade path.
+ */
+contract PreviousRegistrar {
+    mapping (bytes32 => entry) public entries;
+
+    struct entry {
+        uint status;
+        Deed deed;
+        uint registrationDate;
+        uint value;
+        uint highestBid;
+    }
+
+    // Only exists for testing purposes
+    function register (bytes32 _hash) {
+        entry e = entries[_hash];
+        e.status = 2;
+        e.registrationDate = now;
+    }
+}
+
 /**
  * @title Registrar
  * @dev The registrar handles the auction process for each subnode of the node it owns.
  */
 contract Registrar {
     AbstractENS public ens;
     bytes32 public rootNode;
+    PreviousRegistrar public previousRegistrar;
```

Comment:
Well I can put the full code of the ropsten registrar on a separate file, but this is supposed to be more general than just a ropstein 1.0 > ropstein 2.0. It could also be a migration path for the main net, where we could first publish a temporary code with a 4 week auction period and then after a month migrate to a 1 week auction, which is why I removed *initialAuctionPeriod*, and it could be the migration path to the permanent registrar and any other security upgrade necessary